bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/discovery/HomekitBridgedAccessoryDiscoveryService.java

```diff
@@ -0,0 +1,87 @@
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.homekit.internal.dto.Accessory;
+import org.openhab.binding.homekit.internal.handler.HomekitBridgeHandler;
+import org.openhab.core.config.discovery.AbstractThingHandlerDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * Discovery service component that publishes newly discovered bridged accessories of a HomeKit bridge
+ * accessory. Discovered devices are published as Things with thingUID based on accessory ID (aid) of type
+ * {@link org.openhab.binding.homekit.internal.HomekitBindingConstants#THING_TYPE_BRIDGED_ACCESSORY} .
+ *
+ * @author Andrew Fiddian-Green - Initial Contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class)
+public class HomekitBridgedAccessoryDiscoveryService
+        extends AbstractThingHandlerDiscoveryService<HomekitBridgeHandler> {
+
+    private static final int TIMEOUT_SECONDS = 10;
+
+    public HomekitBridgedAccessoryDiscoveryService() {
+        super(HomekitBridgeHandler.class, Set.of(THING_TYPE_BRIDGED_ACCESSORY), TIMEOUT_SECONDS);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        thingHandler.registerDiscoveryService(this);
+    }
+
+    @Override
+    public void dispose() {
+        thingHandler.unregisterDiscoveryService();
+        super.dispose();
+    }
+
+    @Override
+    public void startScan() {
+        if (thingHandler instanceof HomekitBridgeHandler handler) {
+            discoverBridgedAccessories(handler.getThing(), handler.getAccessories().values());
+        }
+    }
+
+    private void discoverBridgedAccessories(Thing bridge, Collection<Accessory> accessories) {
+        String bridgeUniqueId = thingHandler.getThing().getConfiguration()
+                .get(CONFIG_UNIQUE_ID) instanceof String uniqueId ? uniqueId : null;
+        if (bridgeUniqueId == null) {
+            return;
+        }
+        accessories.forEach(accessory -> {
+            if (accessory.aid instanceof Long aid && aid != 1L && accessory.services != null) {
+                ThingUID uid = new ThingUID(THING_TYPE_BRIDGED_ACCESSORY, bridge.getUID(), aid.toString());
+                String uniqueId = STRING_AID_FMT.formatted(bridgeUniqueId, aid);
+                String label = THING_LABEL_FMT.formatted(accessory.getAccessoryInstanceLabel(), uniqueId);
+                thingDiscovered(DiscoveryResultBuilder.create(uid) //
+                        .withBridge(bridge.getUID()) //
+                        .withLabel(label) //
+                        .withProperty(CONFIG_ACCESSORY_ID, aid.toString()) //
+                        .withProperty(PROPERTY_UNIQUE_ID, uniqueId).withRepresentationProperty(PROPERTY_UNIQUE_ID)
```

Comment:
This could be a problem for manually created things because they will not have this property set? I think it also might need to be provided in the XML Thing definition:
```xml
<representation-property>uniqueId</representation-property>
```

I'll double-check this by testing.